### ISSUE TITLE
Missing definition for io.k8s.apimachinery.pkg.apis.meta.v1.Patch in API

### ISSUE THREAD

**Issue by masroorhasan** [kubernetes/kubernetes#54332]
/kind bug

Hello,

I'm working on a generated client from the swagger API. Something I noticed was the missing schema definition of `io.k8s.apimachinery.pkg.apis.meta.v1.Patch` in the PATCH request of `/api/v1/namespaces/{namespace}/configmaps/{name}`. The `parameters` of this request refers to the schema as follows:
`
schema": {
"$ref": "#/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.Patch"
}
`

Compare this to the PUT request of `/api/v1/namespaces/{namespace}/configmaps/{name}` where the `parameters` refers to the schema as:
`
schema": {
"$ref": "#/definitions/io.k8s.kubernetes.pkg.api.v1.ConfigMap"
}
`

However, in the latter `io.k8s.kubernetes.pkg.api.v1.ConfigMap` can be found with a properly defined schema as
`
"io.k8s.kubernetes.pkg.api.v1.ConfigMap": {
"description": "ConfigMap holds configuration data for pods to consume.",
"properties": {
"apiVersion": {
"description": "APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources",
"type": "string"
},
"data": {
"description": "Data contains the configuration data. Each key must consist of alphanumeric characters, '-', '_' or '.'.",
"type": "object",
"additionalProperties": {
"type": "string"
}
},
"kind": {
"description": "Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds",
"type": "string"
},
"metadata": {
"description": "Standard object's metadata. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata",
"$ref": "#/definitions/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta"
}
}
}
`
where as the schema definition of `io.k8s.apimachinery.pkg.apis.meta.v1.Patch` cannot be found in the API.

Let me know if this is not a bug or a version mismatching issue and will close the issue ASAP. 

Thanks.

**What happened**:
I noticed while browsing the swagger API  that its missing the definition for :`io.k8s.apimachinery.pkg.apis.meta.v1.Patch`.

**What you expected to happen**:
A definition of like the following `"io.k8s.apimachinery.pkg.apis.meta.v1.Patch": { .. }`.

**How to reproduce it (as minimally and precisely as possible)**:
Navigate to the proxy API for your K8s cluster to the following url `http://127.0.0.1:8001/swagger.json`.

**Environment**:
- Kubernetes version (use `kubectl version`): 
Client:
`Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.4", GitCommit:"793658f2d7ca7f064d2bdf606519f9fe1229c381", GitTreeState:"clean", BuildDate:"2017-08-17T17:03:51Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}` 

Server:
`Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-07-26T00:12:31Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}`

- Cloud provider or hardware configuration**:
Running a minikube cluster with version: `minikube version: v0.21.0` 

- OS (e.g. from /etc/os-release):
Mac OS Sierra Version 10.12.6

- Kernel (e.g. `uname -a`):
`Darwin Masroors-MacBook-Pro.local 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64`

**Comment by dims** [comment]
cc @mbohlool

**Comment by lavalamp** [comment]
@mbohlool to decide priority/next steps.

**Comment by mbohlool** [comment]
I can see `io.k8s.apimachinery.pkg.apis.meta.v1.Patch` in the latest spec but the schema is empty that can make problem for some generators. What do you generated from the spec? We do have kubernetes-client org with some clients already generated there and also kubernetes-client/gen repo that can be used for generating new clients. The gen repo has a pre-processing script that make the spec swagger-codegen friendly and one of the things it changes is the Patch schema. You may want to take a look at it.

**Comment by masroorhasan** [comment]
Thanks @mbohlool - looks like I missed the kubernetes-client/gen repo. My approach was to directly use swagger code gen to generate c# client from json schema `http://127.0.0.1:8001/swagger.json`. 

Thanks for your response and closing this issue :).

**Comment by mbohlool** [comment]
We also have a [c# client](https://github.com/kubernetes-client/csharp) you may want to check out. It is generated using autorest instead of swagger-codegen.

**Comment by masroorhasan** [comment]
@mbohlool found the c# client while looking through all the k8s clients in that repo. I ended up installing the pre-release package from nuget and unfortunately as I tried to integrate it with my .NET 4.6.1 project I ran into a bunch of issues while using it, see: [11100](https://github.com/dotnet/corefx/issues/11100) and [7522](https://github.com/dotnet/corefx/issues/17522).

I ended up creating a .NET framework version of c# k8s client using combination of swagger code-gen and k8s client gen preprocessor. See here: [Kubernetes.DotNet](https://github.com/masroorhasan/Kubernetes.DotNet).

Let me know if those issues related to c# client can be solved (dealt with separately) as I wouldn't mind switching to the official c# client since it seems more stable than mine.
